New helm chart for kanister-operator

depohmel opened the pull request:
Incubator chart: https://kanister.io/
Quick start intrustion: https://github.com/kanisterio/kanister

Commit:
New helm chart for kanister-operator

Commit:
Merge pull request #1 from kanisterio/kanister-operator-helm-chart

Incubator chart: https://kanister.io/
Quick start intrustion: https://github.com/kanisterio/kanister

Commit:
Trailing whitespaces fix

paulczar commented on incubator/kanister-operator/README.md:
```diff
@@ -0,0 +1,62 @@
+Parameter | Description | Default
+--- | --- | ---
+`rbac.create` | all required roles and SA will be created | `true`
+`rback.service_account`| porvided service account name will be used | `None`
```
should be `rbac.service_account`

depohmel replied:
@paulczar thanks.  /done

Commit:
making rbac.service_account consistent across all files

depohmel commented:
/assign @mgoodness 
/approve

prydonius commented:
@depohmel if the chart meets all the technical requirements and documentation requirements in https://github.com/kubernetes/charts/blob/master/CONTRIBUTING.md#technical-requirements this can be put straight into the stable repository. Can't see anything obvious violating those requirements.

depohmel commented:
thanks, @prydonius 
Can I do move as part of this PR or will it be better to create a new one?

prydonius commented:
@depohmel ideally in this PR, so we can avoid it being in incubator at all.

Commit:
miving to stable per suggecstion in PR's comments

depohmel commented:
@prydonius, thanks.  Just pushed a move.

Commit:
updating README incubator->stable

depohmel commented:
/assign @prydonius

Commit:
Updating kanister-operator release version

tdmanv commented:
@prydonius thanks for all the help. Is there anything left we can do to get this merged?

Commit:
fixing incorrect value name.

unguiculus commented on stable/kanister-operator/Chart.yaml:
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
+description: Kanister-operator Helm chart for Kubernetes
+name: kanister-operator
+version: 0.2.0
```
Add `appVersion` and `sources`. Use Github username for maintainers.

unguiculus reviewed (changes requested):
In general, please follow our guidelines:
https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md

Make sure you use namespaced templates. Ideally you use `helm create` with Helm 2.8+ and take the `_helpers.tpl` you get.

Note that our RBAC best practices have been revised. Please update accordingly.

https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/rbac.md

Go over readme and notes again. From quickly skimming over them I noticed a few typos.

Commit:
Adding appVersion, source and github usernames for maintainers

Commit:
updating _helpers.tpl to use namespace functions

updating all function calls

Commit:
bring rbac configuration to the latest best practices

Commit:
fixing typos

depohmel commented:
@unguiculus , thanks for the review.

Commit:
adding missing new line

Commit:
tables->table there is only one table

Commit:
default image tag 0.2.0 -> v0.2.0

unguiculus commented on stable/kanister-operator/templates/rbac.yaml:
```diff
@@ -3,50 +3,52 @@ kind: ClusterRole
   - "*"
   verbs:
   - "*"
+{{- if .Values.serviceAccount.create }}
```
Please put the service account into its own file.

unguiculus commented on stable/kanister-operator/templates/_helpers.tpl:
```diff
@@ -0,0 +1,51 @@
+{{- define "kanister-operator.helmLabels" -}}
+heritage: {{ .Release.Service }}
+release: {{ .Release.Name }}
+chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
```
Use the `chart` template.

unguiculus commented on stable/kanister-operator/templates/deployment.yaml:
```diff
@@ -0,0 +1,22 @@
+apiVersion: extensions/v1beta1
```
`apps/v1beta2`

unguiculus commented on stable/kanister-operator/README.md:
```diff
@@ -0,0 +1,63 @@
+This chart bootstraps a kanister-operator deployment on a [Kubernetes](http://kubernetes.io) cluster using the [Helm](https://helm.sh) package manager.
+
+## Prerequisites
+  - Kubernetes 1.7+ with Beta APIs enabled
```
1.8+

We officially support the current and previous minor versions of Kubernetes.

unguiculus reviewed (changes requested):
Almost there.

Commit:
addressing requested review changes

serviceAccount is getting it's own file
minimal version 1.8+
chart template into chart label
apps/v1beta2 for Deployment

depohmel commented:
@ultimateboy could you please do another look?

unguiculus commented:
/ok-to-test

unguiculus commented:
/lgtm

depohmel commented:
Thanks @unguiculus!
